schema_utils.py

```diff
@@ -387,3 +388,86 @@ def is_valid_email(obj):
             bool. Whether the given object is a valid email.
         """
         return bool(re.search(r'^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$', obj))
+
+    @staticmethod
+    def is_valid_asciimath_expression(obj, algebraic=True):
+        """Checks if the given obj (a string) represents a valid algebraic or
+        numeric expression. The expression should be in the ASCIIMath format.
+        More info: http://asciimath.org/
+
+        Args:
+            obj: str. The given expression.
+            algebraic: bool. True if the given expression is algebraic
+                else numeric.
+
+        Returns:
+            bool. Whether the given object is a valid expression.
+        """
+
+        if not expression_parser.is_valid_expression(obj):
+            return False
+
+        expression_is_algebraic = expression_parser.is_algebraic(obj)
+        # If the algebraic flag is true, expression_is_algebraic should
+        # also be true, otherwise both should be false which would imply
+        # that the expression is numeric.
+        return not algebraic ^ expression_is_algebraic
+
+    @staticmethod
+    def contains_valid_placeholders(obj):
```

Comment:
I think you can drop this. See a previous comment.

Reply:
Done.